Dockerfile

```diff
@@ -279,5 +279,8 @@ VOLUME /var/lib/docker
 ENTRYPOINT ["hack/dind"]
 
 FROM dev AS final
+
+ENV TEST_FILTER=TestBuildWithEmptyLayers
```

Comment:
FYI unrelated to the underlying issue, but i realized that because this is defined here in the Dockerfile (for debugging purposes) it is still overridden by the Makefile which does `docker run -e TEST_FILTER`

Reply:
`-e TEST_FILTER` takes the value from the current environment (if set), so that should work?

Reply:
@thaJeztah apparently even if it's not set.

```
# cat Dockerfile
FROM busybox
ENV FOO=bar
CMD ["/bin/env"]
# docker build -t foo .
[+] Building 0.1s (5/5) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 31B                                                                                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/busybox:latest                                                                                                                                                                        0.0s
 => CACHED [1/1] FROM docker.io/library/busybox                                                                                                                                                                                          0.0s
 => exporting to image                                                                                                                                                                                                                   0.0s
 => => exporting layers                                                                                                                                                                                                                  0.0s
 => => writing image sha256:ff24d519ee276e63e55816970c12c9054b92c89afe4cd6ede254f6426cb7c55e                                                                                                                                             0.0s
 => => naming to docker.io/library/foo                                                                                                                                                                                                   0.0s
# docker run -it --rm foo
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=720ff00a9b1f
TERM=xterm
FOO=bar
HOME=/root
# env | grep FOO
# docker run -it --rm -e FOO foo
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=1117377f660c
TERM=xterm
HOME=/root
```